runelite-client/src/main/java/net/runelite/client/plugins/huntertracker/HunterTrackerPlugin.java

```diff
@@ -0,0 +1,208 @@
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.huntertracker;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import java.awt.image.BufferedImage;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import javax.imageio.ImageIO;
+import javax.swing.SwingUtilities;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.VarPlayer;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.UsernameChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.NavigationButton;
+import net.runelite.client.ui.PluginToolbar;
+
+@PluginDescriptor(
+		name = "Hunter Tracker"
+)
+@Slf4j
+public class HunterTrackerPlugin extends Plugin
+{
+	@Inject
+	private PluginToolbar pluginToolbar;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private HunterWorld hunterWorld;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private HunterTrackerConfig config;
+
+	private HunterTrackerPanel panel;
+
+	private NavigationButton navButton;
+
+	private WorldPoint lastTickLoc;
+
+	private int tickCount = 0;
+
+	@Provides
+	HunterTrackerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(HunterTrackerConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		BufferedImage icon;
+		synchronized (ImageIO.class)
+		{
+			icon = ImageIO.read(getClass().getResourceAsStream("hunter.png"));
+		}
+
+		panel = new HunterTrackerPanel(client, itemManager, configManager, config, hunterWorld);
+
+		navButton = NavigationButton.builder()
+				.tooltip("Hunter Tracker")
+				.icon(icon)
+				.panel(panel)
+				.priority(4)
+				.build();
+
+		pluginToolbar.addNavigation(navButton);
+
+		updatePanel();
+	}
+
+	@Subscribe
+	public void onUsernameChanged(UsernameChanged e)
+	{
+		updatePanel();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		pluginToolbar.removeNavigation(navButton);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick t)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			// Reset the ticks if you aren't logged in due to VarP initialization
+			tickCount = 0;
+			lastTickLoc = null;
+			return;
+		}
+
+		WorldPoint loc = lastTickLoc;
+		lastTickLoc = client.getLocalPlayer().getWorldLocation();
+		if (loc == null || loc.getRegionID() != lastTickLoc.getRegionID())
+		{
+			return;
+		}
+
+		// It takes a bit for the VarPs to initialize, if you login at the birdhouses give it a few to update
+		if (tickCount <= 10)
```

Comment:
Does this really take this long? I have never seen any varp not synced by after the first game tick (which fires a cycle after npc update)

Reply:
It may be worth looking into a more general 'post login' event since this has been an issue before with the daily indicators plugin. Look at what it does.